### PR TITLE
cache ci slow steps

### PR DESCRIPTION
**Pull request by chrisns**
brings builds from ~4mins down to <1m for routine stuff, can be tweaked to be a bit more refined if needed

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,47 +9,83 @@ on:
 jobs:
   test:
     runs-on: ubuntu-latest
+
     strategy:
       matrix:
         node-version: [8.x, 10.x, 12.x]
```
Comment (robertsLando) [inline .github/workflows/ci.yml:15]: Should we add 14.x?

**Comment by robertsLando** [comment]
Love this improvments ❤️

File: .github/workflows/ci.yml (hunk 1 above)
Comment (chrisns) [reply .github/workflows/ci.yml:15]: ... and drop `8`? ;)

Comment (robertsLando) [reply .github/workflows/ci.yml:15]: Definitely :)

Comment (chrisns) [reply .github/workflows/ci.yml:15]: in #416

**Pull request merged**